### ISSUE TITLE
Improved handling of the `distance_to_colony` calculation when colony of bird changes

### ISSUE THREAD

**Issue by stijnvanhoey** [inbo/gulltracking#11]
Currently, the [catch location](https://docs.google.com/spreadsheets/d/168lV0h4f0uljItStoNX16j8stjJg640ozrgzHW-5d9w/edit#gid=1087884704) is assumed to be the colony of the bird, as described [here](https://github.com/inbo/bird-tracking-etl/blob/master/vignettes/bird-tracking-etl.Rmd#output-format). 

It was assumed the colony does not change in time. However, Jasmin (858) was originally part of Zeebrugge, but in moved in 2016 to Vlissingen. Therefore, the distances are currently wrong. 

Correcting this, requires adaptation of the [`enrich_data` function](https://github.com/inbo/bird-tracking-etl/blob/master/R/enrich.R#L309) and more specific, the [`add_dist_to_colony` function](https://github.com/inbo/bird-tracking-etl/blob/master/R/enrich.R#L197) OR making sure the [join of the logs with the metadata](join_tracks_and_metadata) is adapted to tackle this change in colony. Hence, it would also require an adaptation of the way the colony location (i.e. colony) is defined in the metadata. 

Some questions:
* Should we change the assumption of the `catch_location` as colony?
* How would we represent the colony change inside the metadata?
* Do we have the *exact* date of the move towards another colony and could we include this in the data flow in order to take this into account in the data?

**Comment by peterdesmet** [comment]
> Hence, it would also require an adaptation of the way the colony location (i.e. colony) is defined in the metadata.

We've previously thought about storing tracking periods, rather than birds in the metadata. It's maybe good to revisit this. I'm also curious how this is stored at UvA-BiTS (as I still think it's a bit weird we're storing our tracking metadata in Google Spreadsheet, outside the UvA-BiTS system).

**Comment by hmatheve** [comment]
In the UvA-BiTS system they make use of tracking sessions. Basically, it says when a tracker has been deployed on a specific bird. Until now we don't have any event where a tracker has been re-used.

These tracking sessions might be of use for giving updates for it's breeding (sub-)colony. We could add an extra record for one bird each time it switches nesting location. The timing is a bit tricky as it would result in an abrupt increase/decrease of distances to nest. If we would set this date by default on the first of January of the previous winter, the bird is deep down in it's wintering area and thus the distance to nest is altered proportionally only little.

So:
- catch_colony is NOT 'breeding_colony' (although in >90% it is the same)
- in the metadata the breeding_colony can be filled in by taking the date of the new track session into account?
- would be 01-01-20xx (see above)

**Comment by stijnvanhoey** [comment]
Actually, we do already have an event where a tracker has been re-used, if you look to the [meta-data](https://docs.google.com/spreadsheets/d/168lV0h4f0uljItStoNX16j8stjJg640ozrgzHW-5d9w/edit#gid=1087884704), the tracker with `device_info_serial` 5338 has been deployed to bird `Roy` from 2016-05-20 till 2016-05-30 and to bird `Fay` from 2016-05-31 on. Based on this switch, we adapted the ETL procedure earlier. The join of metadata to the logs uses both the `device_info_serial` and the time period (`tracking_started_at` and `tracking_ended_at`). See issue #9 for more information.

With adequate alterations to the R-package, the extension of the metadata table with a `breeding_colony` field and corresponding coordinates (latitude/longitude) can be taken into account for the distance calculations. 

The `tracking_started_at` and `tracking_ended_at` is not really comparable to the concept of a tracking session, as it defines the deployment period for a given bird. However, technically I do think (testing would be required!) this would work (as periods are taken into account in the join). It would result in multiple lines with the same bird name, but different `tracking_started_at` and `tracking_ended_at` together with any other required adaptation (e.g. `breeding_colony`). The join would still work and the distances would be calculated from the relevant `breeding_colony` coordinates for the corresponding period/`device_info_serial` combination. As such, the concept of tracking sessions can - relatively easy -  be included in the current ETL (however making the metadata more verbose). 

However, before starting the implementation using the altered metadata google-spreadsheet, it is maybe worthwhile considering the suggestion of @peterdesmet. How is this actually stored at UvA-BiTS? Would be interesting to make the comparison to choose the most relevant option.

**Comment by peterdesmet** [comment]
This applies to the [BirdTrackingEtl](https://github.com/inbo/gulltracking/releases/tag/bt-etl) version of the package, which we no longer maintain. Closing with `wontfix`.
